### ISSUE TITLE
building with --mode outputs a blank window

### ISSUE THREAD

**Issue by Bbansjkl** [nklayman/vue-cli-plugin-electron-builder#164]
**Describe the bug**
When the argument `--mode <environment>` is used with electron:build to change the environment of the Vue application, the built electron app is completely blank.  
Serving with `--mode <environment>` works fine.  
Whether the router is set to history mode or not doesn't seem to be the culprit.

**To Reproduce**
create a new vue app:
```
vue create electronEnvBuildTest
>Manually select features
>Babel, Typescript, Router, Vuex, Linter/Formatter
? Use class-style component syntax >Y
? Use Babel alongside Typescript ... > Y
? Use history mode for router? ... > N
? Pick a linter ... > TSLint
> Lint on Save
> Dedicated config files

cd electron-test2
vue add electron-builder
? Choose Electron Version
> ^2.0.0 //Also happens with 3.0.0

```
Open ./src/background.ts and fix line 12: `let win: BrowserWindow | null;`  
Add a file .env with the content `VUE_APP_TEST=Default  
Add a file .env.foo with the content `VUE_APP_TEST=Foo  
Open ./src/views/Home.vue and add `public env = process.env.VUE_APP_TEST;` inside the `Home` class, add `{{env}}` somewhere in the `template` section.
Open ./package.json and add `"electron:build:foo": "vue-cli-service electron:build --mode foo",` to the 
`scripts` section.

Or simply clone my repo linked below

Run the command `npm run electron:build:foo` and open the resulting electron application

**Expected behavior**
The Vue application with the correct environment applied should display

**Screenshots**
![grafik](https://user-images.githubusercontent.com/39167840/50145023-d73b5400-02b0-11e9-82fa-1c23c260e110.png)

**Environment (please complete the following information):**
 - OS and version: Windows 7 Pro (Version 6.1 Build 7601: Service Pack1)
 - node version: v8.12.0
 - npm version: 6.4.1
 - yarn version (if used): -
 - vue-cli-plugin-electron-builder version : 1.0.0-rc.9
 - electron version: 2.0.15
 - other vue plugins used: -
 - custom config for vcp-electron-builder: -
 - (if possible) link to your repo: https://github.com/Bbansjkl/electron-build-env-test

**Additional context**

**Comment by nklayman** [comment]
Try setting `process.env.NODE_ENV` to `production`.

**Comment by Bbansjkl** [comment]
Adding `NODE_ENV=production` to `.env.foo` works, thanks!

Now this breaks `electron:serve --mode foo` instead, though. So I'd probably need more .env files for all the cases. 

Still weird that it doesn't work without. I guess that has to do with the way electron:build and electron:serve work - any workaround so that they check which mode they use depending on NODE_ENV instead of the command used? Or is that not in scope of the vue plugin and should be raised on the electron-builder repository instead?

**Comment by nklayman** [comment]
That is caused by the background.js checking for node_env. I will fix that later today. Thanks for reporting the issue!

**Comment by nklayman** [comment]
fixed with 1.0.0-rc.10. Now, it will check for `WEBPACK_DEV_SERVER_URL` instead of `process.env.NODE_ENV`. To use this without re-invoking the generator, replace
```javascript
  if (isDevelopment) {
    // Load the url of the dev server if in development mode
    win.loadURL(process.env.WEBPACK_DEV_SERVER_URL)
...
```
with
```javascript
  if (process.env.WEBPACK_DEV_SERVER_URL) {
    // Load the url of the dev server if in development mode
    win.loadURL(process.env.WEBPACK_DEV_SERVER_URL)
...
```
